contrib/nydusify/pkg/utils/utils_test.go

```diff
@@ -0,0 +1,196 @@
+// Copyright 2023 Alibaba Cloud. All rights reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+package utils
+
+import (
+	"archive/tar"
+	"compress/gzip"
+	"io"
+	"os"
+	"strings"
+	"testing"
+
+	"github.com/opencontainers/go-digest"
+	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
+	"github.com/stretchr/testify/assert"
+)
+
+func makePlatform(osArch string, nydus bool) *ocispec.Platform {
+	var platform *ocispec.Platform
+	if osArch == "" {
+		platform = &ocispec.Platform{
+			OS:           "",
+			Architecture: "",
+		}
+	} else {
+		platform = &ocispec.Platform{
+			OS:           strings.Split(osArch, "/")[0],
+			Architecture: strings.Split(osArch, "/")[1],
+		}
+	}
+	if nydus {
+		platform.OSFeatures = []string{ManifestOSFeatureNydus}
+	}
+	return platform
+}
+
+func makeDesc(id string, platform *ocispec.Platform) ocispec.Descriptor {
+	return ocispec.Descriptor{
+		MediaType: ocispec.MediaTypeImageManifest,
+		Digest:    digest.FromString("manifest-" + id),
+		Size:      10,
+		Platform:  platform,
+	}
+}
+
+func TestIsSupportedArch(t *testing.T) {
+	var arch string
+	arch = PlatformArchAMD64
+	assert.Equal(t, IsSupportedArch(arch), true)
```

Comment:
Why not `require.Equal(t, IsSupportedArch(arch), PlatformArchAMD64)`

Reply:
Fixed.